metrics-core/src/main/java/com/codahale/metrics/LockFreeExponentiallyDecayingReservoir.java

```diff
@@ -64,21 +65,18 @@
         private void update(long value, long timestampNanos) {
             double itemWeight = weight(timestampNanos - startTick);
             double priority = itemWeight / ThreadLocalRandom.current().nextDouble();
+            long currentCount = count.get();
+            if (currentCount < size) {
+                addSample(priority, value, itemWeight);
+            } else if (values.firstKey() < priority) {
```

Comment:
Minor: could replace else-if with `||` since it's the same code for both blocks

Reply:
Good call!